dev/KaairaGupta/winequality/data_visualisation.ipynb

```diff
@@ -0,0 +1,1421 @@
p6DggJkVrZD47enYb9Ior7dAUDy/jPovD1wK3E+Bgi9vQXnrA2ZqG9+p2H/SAB8m9YlbscxAC6ficeznj+u/t4Oib3k9TO+jOtnrPX6mb4vCifr9RPg0u4T5KdfvQ5C/nMbcd/vrJwEriGyb0d2fLcVgDMHTuHh7YFPoG+pcgab672zi44rf26r85Bu7Fm3m9Q4y3U0I/Vypcdclpv6RnLou98BuHggCldvdzzt7GenfztU+HvsoTN4h1i+02SnZhB3+Czm/OpxnmvWN5LD1nzirpJP4rHzXLZ+jylGgounKwDOHq4Y0rMwF5grNeYqZ66kn6u7CNS3eV0PKPllsj2wWggRDdwLLBFCDC1/gsXdEA04IcQoIcRhIcQhIcRn1sXdhBA7hRBny+iNcxVCfCyEOCKEOCCE6GldPkYI8Y0Q4mdgg22vnRDCTQix2vpZXwFuNu/XTwixSwix37q9p3X560KI49ZtKvz2lUuwHkNcauHr3Lg0XKyNsMIyIXoMsTZl4tNwCSlexq1+AN4tG5G+P6pwWdNZ99N9/2JC7unC6TeLD1OqLC4hevLiik5WxvjUUrHqQvzIs8nZGJ+GS4hfsTIBQ28n+YcdlRtsOelC9BjjbXJLSEVXIrfqzCVYjzHWtm7S0AUX/3/XBRevv7z4VHQhfhgT0ohb+hORe5fS4dAKTJk5XN5mudDmnIjB19oQ9BvcGZfQ6vOFu6bTBfthLHaspOJcg/a5krRBfuQnFN2JLkhIQRtY9v7ifc8Asn/f44jQrosmwB9TUlLha1NSCtqAgDLLewweRO4uyygCp7ohmNMv4/PsTAI+/QCfWdMQrq6VHvO1uAf7km2zj+XEp+ERXLyR4OLriTEjB2myfFvKtimTejyGRgMtx79/28Z41vPHw0H7qGuwnlyb2A12rp+uIXpyS1w/Xa8zPt9OzTEmp5NzLqFiAi4n32B9YeMLIC0hFd8g+7E/Mn8Si/d+RGh4XTZ8YumlCg4LxaOOJ3NWv8wrv8yny909HBG2Xd7BejJs6iojIQ3voNKN0Ss0Tlra3N2FqK2HHRFeuXnZycfrKvmUtPfTDfiF1+XJPYt4ZP3rbHzpM7jB/tC1g+wBmgohwoQQOmA4UGyctZQyTErZSErZCPgWmCClLD2MqpxqfQNOCHELMAfoJaVsA1wZLxMCdAHuBF63s+lEACllK+AB4FMhxJWr4W3AaCllrxLbPA7kSClbA68CkdYY/IFngT5SyghgL/CUEEKPZVzsLdZt7I7BEkI8IoTYK4TYu8ZQzmez7HXuXk93mc2BrnV3oe2HUznx3KfFet5Ov/YV2yImEv/dDhqO61++uP4pYSehEicle0VsywhnJ/z6tSflp10VHNy/VTrwGnW+va66sV9GW8cDff8O7Lt1AnvbPozG3RX/e7oBcOapJYSMHUDr9W+i9XDFbCyojOhvTNcz+KMmsX/w2y3q1rEN3nf3J3Xhh5UbU3lcxzF0hS6iLe6DB5KxeLllgVaLc7OmZP/vJ5JHP4o05OI56oFKDPb62Dvm5XWcF66UOLz4Z3R1PBi6/lVuGduP1KPnkY7qSfiH18+S+ZUl9K7bq6z3Da6vbq5YNmMRkzo+RFxULJ0GW4a3a5w0hLVswoKxr/LGyJcZ+sS9BIeFVGrMZbJ76JRdD3fMHcv5P08Qs+dkJQb1z9k/FVz/yblx99YkHjvPux0msWLgbPq/PBqdp9u1N6zBpFlWys9VP1PKAmASltkl/wa+llIeE0I8JoR4rDLzvREmMekFfCulTAGQUqZZT1o/SCnNwHEhRJCd7boA71m3OSGEOA80s67bKKVMs7NNN+D/rNscFkJcubXTCcv0on9YP1sH7AIygFxghRDiV+AXewlYx9suA1gXNLxcX6/y4tNwCy3qBXEN1ZOXcKl0mbp+pF8pE1JURjhpaffRU8R/t4PENfbvVMf97w8iv3iaqPl2HwWsUHlxqcV6YHQhfqXziUvFxSZnXYievISi6vLt1Y6sI+fIT6m64R72GONT0YXY5BbsR36Cvd2sesqLT0VX17Zu9IXDIG3LuIT6k2l97WLtffPp2prcmCQKUjMASFuzG+/2N5Hy3XYMUbEcH/4KAK6NQ/DtE+mQfG4ExvhUdMWOlZq1z5VkSkjBObiox8op2B9TUmqpcrpmYQS+PIW4R5/FfDmz1PqqYk5KRhsYWPhaG+iPKaX08CinJo3xnTWd1KeewZxhOWZMScmYkpPJP34CAMNv2/EaWTUNuJtH9+GmB3sCkHLoLB42+5h7iJ6cxPRi5XPTMtF5uyO0GqTJjEeInhzreT0/y8Dv04oeOblv19tklnjep7LkxqfhahO7m53rZ258Gq51ba6xIaXL2CO0GoLv6MAffWdXXMDXoc+oAfQc3heAs4ej8LO5nuqD/UhPKjt2aTaz++cd3PHoULZ/s4VL8akcTsskz5BHniGPE38dp8HNjUg4F1/peQB0GNWXyOGW/Sz28Fm8berKO1hPZlK63e26P3k3HnovvppVjW7eYHlurZ01nzg7+WSVkY89bYZ1Y+eSnwG4dD6R9AvJ+DcJIe5Q9ZmwqcJVzSyUSCnXAGtKLLM7m5mUckxFfW6t74HDcl/GXq3mlShjb7uyXG1Qu73PElgafW2tPy2klOOtLfeOWKYWHQqsu8r7/iOXD5zBvXEwbg0CEM5agod2Jmn9vmJlktbvI3SYpbejTmQ4+Zk55FlPFC3ffpSs07FEf1B8Rib3sODC3wP7R5J92jHPj2UejMK1cQguDQIRzk4EDL2dtA3FG5apG/YSeF8PALwimmLKzCHf5sQXeFeXajd8EiD70Glcw0LQ1bfkph/ShfSNVfNw+z+RdTAKt7AQXKzx+w/pQtr6vcXKXFq/h4Bh3QHwjGhKgbVu8mJT8IpshsZNB0CdLq3IOW2Zac7Zz/q8ixDUm3IviSs3OC6pWi7rYBSuNnWmH9KFSxuqz5DC8so9ehLnhnVxqhsEzk54DuxB9m+7i
```

Comment:
Did you mean `0.5, 0.33 and 0.25 proportions` for `true`? That's what the code does as far as I can tell. Eg. when `i = 2` it looks like you are retaining twice as many `false` cases as `true`. It would be great to use these same proportion numbers in the output below rather than `1 proportion`, `2 proportion`, etc.